File: cpp/test/graph/test_dfs.cc
#include <vector>

#include <gtest/gtest.h>

#include "../../graph/dfs.cc"

TEST(DFSTest, TestSingleNode) {
  vector<Node> graph;
  graph.emplace_back(10);
  ASSERT_EQ(DFS(1, graph, 0, 10), 0);
  ASSERT_EQ(DFS(1, graph, 0, 11), -1);
}

File: python/test/graph/test_dfs.py
import unittest

from graph.dfs import Node, DFS

class DFSTest(unittest.TestCase):
    def test_single_node(self):
        n = Node(1)
        self.assertIs(DFS(n, 1), n)
        self.assertIs(DFS(n, 0), None)

